src/vehicle.cpp

```diff
@@ -2513,18 +2514,36 @@ void vehicle::print_fuel_indicator (void *w, int y, int x, bool fullsize, bool v
 
 void vehicle::coord_translate (int reldx, int reldy, int &dx, int &dy)
 {
-    tileray tdir (face.dir());
-    tdir.advance (reldx);
-    dx = tdir.dx() + tdir.ortho_dx(reldy);
-    dy = tdir.dy() + tdir.ortho_dy(reldy);
+    coord_translate(face.dir(), reldx, reldy, dx, dy);
 }
 
+// rotates coordinates using two shear operations
+// adjusts angles to work in the range -44.9999 to +44.9999 degrees
+// first shears in the dy direction proportional tan(dir)/2
+// then shears in the dx direction proportional to tan(dir)
+// there is certainly a more mathematically accurate way to do this
+// perhaps something like...
+// https://www.ocf.berkeley.edu/~fricke/projects/israel/paeth/rotation_by_shearing.html
 void vehicle::coord_translate (int dir, int reldx, int reldy, int &dx, int &dy)
 {
-    tileray tdir (dir);
-    tdir.advance (reldx);
-    dx = tdir.dx() + tdir.ortho_dx(reldy);
-    dy = tdir.dy() + tdir.ortho_dy(reldy);
+    int quad = 0;
+    while (dir > 45) {
+        dir = dir - 90;
+        quad++;
+    }
+
+    dir = dir * 0.9999999999; // floating point discontinuity at 45 degrees otherwise
+    // rotation by shearing
+    double dy_rate = 2.0 / tan(dir * M_PI / 180.0);
```

Comment:
Theoretically:

`dir` starts out with 0 (for new vehicles for example), multiplying it with `0.9999` doesn't change it.
`tan(0)` is still 0, giving you a beautiful `NaN` in d`y_rate`.